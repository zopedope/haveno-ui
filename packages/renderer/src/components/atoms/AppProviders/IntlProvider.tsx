// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
