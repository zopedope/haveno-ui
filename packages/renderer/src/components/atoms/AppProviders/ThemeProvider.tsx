// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
