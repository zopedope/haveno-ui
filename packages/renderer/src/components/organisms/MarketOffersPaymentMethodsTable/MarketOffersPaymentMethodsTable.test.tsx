// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import type { TMarketOfferPaymentMethod } from "./_types";
import { MarketOffersPaymentMethodsTable } from ".";
import { AppProviders } from "@atoms/AppProviders";

describe("molecules::MarketOffersPaymentMethodsTable", () => {
  it("renders without exploding", () => {
    const { asFragment, unmount } = render(
      <AppProviders>
        <MarketOffersPaymentMethodsTable data={data} />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("renders all columns", () => {
    const { unmount } = render(
      <AppProviders>
        <MarketOffersPaymentMethodsTable data={data} />
      </AppProviders>
    );
    expect(screen.queryByText("Method")).toBeInTheDocument();
    expect(screen.queryByText("Rate Trade Limit")).toBeInTheDocument();
    expect(screen.queryByText("Info")).toBeInTheDocument();
    unmount();
  });

  it("renders cells of method name ", () => {
    const { unmount } = render(
      <AppProviders>
        <MarketOffersPaymentMethodsTable data={data} />
      </AppProviders>
    );
    expect(screen.queryByText("Celpay")).toBeInTheDocument();
    expect(screen.queryByText("ACH")).toBeInTheDocument();
    unmount();
  });

  it("renders cells of rate trade limit ", () => {
    const { unmount } = render(
      <AppProviders>
        <MarketOffersPaymentMethodsTable data={data} />
      </AppProviders>
    );
    expect(screen.queryByText("20 XMR")).toBeInTheDocument();
    expect(screen.queryByText("40 XMR")).toBeInTheDocument();
    unmount();
  });

  it("renders cells of rate trade limit ", () => {
    const { unmount } = render(
      <AppProviders>
        <MarketOffersPaymentMethodsTable data={data} />
      </AppProviders>
    );
    expect(screen.queryByText("USA")).toBeInTheDocument();
    expect(screen.queryByText("Global (AUS, TRY, USD)")).toBeInTheDocument();
    unmount();
  });
});

const data = [
  {
    methodName: "Celpay",
    methodKey: "celpay",
    rateTradeLimit: 20,
    rateTradeLimitCurrency: "XMR",
    info: "USA",
  },
  {
    methodName: "ACH",
    methodKey: "ach",
    rateTradeLimit: 40,
    rateTradeLimitCurrency: "XMR",
    info: "Global (AUS, TRY, USD)",
  },
] as Array<TMarketOfferPaymentMethod>;
