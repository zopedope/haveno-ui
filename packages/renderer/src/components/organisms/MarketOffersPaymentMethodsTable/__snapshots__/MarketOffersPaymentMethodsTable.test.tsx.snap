// Vitest Snapshot v1

exports[`molecules::MarketOffersPaymentMethodsTable > renders without exploding 1`] = `
<DocumentFragment>
  <table
    class="mantine-Table-root __mantine-ref-striped mantine-Table-striped mantine-6859xh"
  >
    <thead>
      <tr>
        <th
          colspan="1"
          style="width: 30px;"
        >
           
        </th>
        <th
          colspan="1"
          style="width: 300px;"
        >
          Method
        </th>
        <th
          colspan="1"
          style="width: 400px;"
        >
          Rate Trade Limit
        </th>
        <th
          colspan="1"
          style="width: 400px; text-align: right;"
        >
          Info
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="width: 30px;"
        >
          <div
            class="mantine-Checkbox-root mantine-16ttirm"
          >
            <div
              class="mantine-qbjb2s mantine-Checkbox-inner"
            >
              <input
                class="mantine-1mvrtfa mantine-Checkbox-input"
                id="methodChecked-0"
                type="checkbox"
              />
              <svg
                class="__mantine-ref-icon mantine-180iq9w mantine-Checkbox-icon"
                fill="none"
                viewBox="0 0 10 7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </td>
        <td
          style="width: 300px;"
        >
          Celpay
        </td>
        <td
          style="width: 400px;"
        >
          <div
            class="mantine-Text-root mantine-67rh7b"
          >
            20 XMR
          </div>
        </td>
        <td
          style="width: 400px; text-align: right;"
        >
          <div
            class="mantine-Text-root mantine-67rh7b"
          >
            USA
          </div>
        </td>
      </tr>
      <tr>
        <td
          style="width: 30px;"
        >
          <div
            class="mantine-Checkbox-root mantine-16ttirm"
          >
            <div
              class="mantine-qbjb2s mantine-Checkbox-inner"
            >
              <input
                class="mantine-1mvrtfa mantine-Checkbox-input"
                id="methodChecked-1"
                type="checkbox"
              />
              <svg
                class="__mantine-ref-icon mantine-180iq9w mantine-Checkbox-icon"
                fill="none"
                viewBox="0 0 10 7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </td>
        <td
          style="width: 300px;"
        >
          ACH
        </td>
        <td
          style="width: 400px;"
        >
          <div
            class="mantine-Text-root mantine-67rh7b"
          >
            40 XMR
          </div>
        </td>
        <td
          style="width: 400px; text-align: right;"
        >
          <div
            class="mantine-Text-root mantine-67rh7b"
          >
            Global (AUS, TRY, USD)
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
